### PR TITLE
Minor initialisation fixes

### PR DESCRIPTION
**Pull request by xverges**
# Background
#### Link to issue 

N/A - As a result of using a broken local development environment that failed to load cellSets, I got several loud errors. This PR prevents them from throwing.
 
#### Link to staging deployment URL 

https://ui-xavier-ui222.scp-staging.biomage.net/

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

* Set the error flag instead of the loading when fetching cellSets or samples returns an error status.
* Set better guards for Continuous Embedding plot and Violin plot

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock
- [x] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Review by aerlaut — DISMISSED** [review]
Changes look good, I've tested it in local. In non-`development` environment, these changes will enable the platform to load without the presence of `cellSets`, as `ContentWrapper` is bypassed. However, in production / staging, I believe there is a guarantee that `cellSets` will be available as data-ingest / pipeline is run?

**Review by kafkasl — DISMISSED** [review]
why would we want to hide errors? this broken local env is temporary as we merge the cellsets changes. I'm worried that hiding these errors will make it harder to debug in the future. Shouldn't the cellsets be always present? if they are not we need to fix the env, not hide errors. Or am I missing something?

**Comment by xverges** [comment]
> why would we want to hide errors?

`cellSets`, specifically their `properties` and `hierarchy` attributes, are loaded asynchronously, so we cannot blindly assume that they are already loaded. The broken environment brought us to this situation, but could have got here by means of a network glitch.

**Review by kafkasl — COMMENTED** [review]
ok, your comment makes sense

File: src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ const generateSpec = (config, plotData) => {
 };
 
 const filterCells = (cellSets, selectedCellSet) => {
-  let newCellSets = cellSets.hierarchy.find(
+  let newCellSets = cellSets.hierarchy?.find(
```
Comment (marcellp) [inline src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js]: This is probably not the place to do this, as there is no guarantee that `filterCells` and `generateData` will re-run once the cell sets become available.

Are you sure this is a problem at all? Can you tell me how you managed to produce the bug? I checked `CategoricalEmbeddingPlot` and it seems that there is a check to ensure this plot does not load unless cell sets are available (see [here](https://github.com/biomage-ltd/ui/blob/master/src/components/plots/CategoricalEmbeddingPlot.jsx#L47-L51)).

Comment (xverges) [reply src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js]: I was looking there also, until I realised that the issue came from https://github.com/biomage-ltd/ui/blob/57d9e8df13e877690e49f9ac06b4598651efb1aa/src/components/data-processing/ConfigureEmbedding/ConfigureEmbedding.jsx#L336

**Comment by xverges** [comment]
The core reason for the failures was that the status of the fetch was not being checked, so `cellSets.error` was not set. I hav undone some of the previous changes. Now looking at the guards in `useEffect()` for the violin plot and the continuous embedding plot, because in this situation they get put the browser into a crazy CPU load.

**Review by ivababukova — APPROVED** [review]

**Pull request merged**